### ISSUE TITLE
Internal microphone not working

### ISSUE THREAD

**Issue by frigis1** [sebanc/brunch#56]
This is on an Acer Spin SP111-31. Internal microphone is not working, but if I plug in a headset with a microphone, the headset mic works fine. Currently using r80 k4.19 stable - r81 k4.19 testing does not work on this laptop.

**Comment by sebanc** [comment]
Hi, the audio in ChromeOS depends on the linux kernel support and generally works the same way as in linux.
Unfortunately, browsing linux forums for your device or similar ones, I found several reports of microphone not working but no solution so it does not seem like I there is something I can do.
Don't hesitate to have a look as well and see if you find something interesting.

I am also interested by the second part of the issue as the recent testing versions are the base for future stable releases. Could you tell me what is the issue with testing versions ? (do not boot, no android apps support...)

**Comment by GGorAA** [comment]
I have the same issue, device HP EliteBook 840 G3

**Comment by frigis1** [comment]
Not sure how much this helps, but mic seems to work fine on Lubuntu 19.10 on a live USB. As for the testing versions, r80 k4.19 testing worked fine, but when I updated to r81 testing using the recommended method, the laptop shuts down a few seconds after the login screen. I also tried installing it from scratch with the same results (shutdown a few seconds after welcome screen). Went back to r80 stable with no problems (microphone aside).

And thank you for the prompt reply. This is amazing work!

**Comment by nicktelindert** [comment]
My internal mic also isn't working at all. Which is really weird because it is linux compatible. The laptop came pre-installed with linux and i have done a lot of distro hopping and never had that problem before. It is a Dell XPS 13 9380

**Comment by nicktelindert** [comment]
I did some testing, the internal mic works when i plugin a headphone(without mic). I will further look into this.

**Comment by nicktelindert** [comment]
I get the internal mic working with the following:
amixer -c0 set "Internal Mic" toggle
amixer -c0 set "Capture" 40%

After reboot the Internal Mic is turned off again.

**Comment by frigis1** [comment]
Interesting, this works for me, too! This setting also resets after reboot for me.

After tinkering a bit, I think I see the problem for my case. There are two toggles, the headset mic and internal mic. Enable one, and the other is automatically disabled. Presumably the headset mic should automatically enable when you plug in headphones, and the internal mic gets enabled when nothing's plugged in.

In my case, the toggle is stuck on headset mic whether or not headphones are plugged in, which means my internal mic is always disabled, but headset mic is fine if I plugged one in. I can manually enable internal mic thanks to the commands you posted, but then the headset mic is disabled.

Thank you for that post, nicktelindert!

**Comment by sebanc** [comment]
> I get the internal mic working with the following:
> amixer -c0 set "Internal Mic" toggle
> amixer -c0 set "Capture" 40%
> 
> After reboot the Internal Mic is turned off again.

Great find !

If you want to automate it, you can create a bootscript by running in crosh shell:
```
sudo mkdir /var/brunch/bootscripts
echo 'amixer -c0 set "Internal Mic" toggle' | sudo tee /var/brunch/bootscripts/sound.sh
echo 'amixer -c0 set "Capture" 40%' | sudo tee -a /var/brunch/bootscripts/sound.sh
sudo chmod 0755 /var/brunch/bootscripts/sound.sh
```

**Comment by jordanvanbergen** [comment]
> I get the internal mic working with the following:
> amixer -c0 set "Internal Mic" toggle
> amixer -c0 set "Capture" 40%
> 
> After reboot the Internal Mic is turned off again.

The first command works on my Lenovo Ideapad Yoga 13 and I thought YES the internal mic is going to work but the second line generates this error:
amixer: Unable to find simple control 'Capture' ,0

**Comment by frigis1** [comment]
Try running:
amixer scontrols
It should come up with 'Master',0 and something else below that - try using that name (not master, but the other one) instead of "Capture."

**Comment by hacker1024** [comment]
@nicktelindert, what do those commands do? They fix it, but why 40%?

**Comment by jordanvanbergen** [comment]
> Try running:
> amixer scontrols
> It should come up with 'Master',0 and something else below that - try using that name (not master, but the other one) instead of "Capture."

I get this output

amixer scontrols
Simple mixer control 'Master',0
Simple mixer control 'Capture',0

So it's Capture but I get:

amixer -c0 set "Internal Mic" toggle
Simple mixer control 'Internal Mic',0
  Capabilities: cvolume cvolume-joined cswitch cswitch-joined
  Capture channels: Mono
  Limits: Capture 0 - 80
  Mono: Capture 0 [0%] [-74.00dB] [on]

and:

amixer -c0 set "Capture" 40%
amixer: Unable to find simple control 'Capture',0

**Comment by frigis1** [comment]
@jordanvanbergen Maybe try: amixer set "Capture" 40% (without -c0)
@hacker1024 The commands seem to enable the internal mic, and then set "capture" (input) to 40% volume, you can set it to 100% volume if you want.

**Comment by sebanc** [comment]
Could you confirm that the issue persist with the new testing release ?

**Comment by jordanvanbergen** [comment]
> Hi, could you confirm that the issue persist with the new testing release ?

Will try today. This will be the 2nd time I am going to upgrade brunch & Chrome OS. I assume I will have to do this: 

sudo chromeos-update -r < path to the ChromeOS recovery image > -f < path to the Brunch release archive >

**Comment by sebanc** [comment]
yes if you want to update both, if you only want to update brunch:
`sudo chromeos-update -f < path to the Brunch release archive >`

**Comment by hacker1024** [comment]
The issue is still there for me.

**Comment by jordanvanbergen** [comment]
> yes if you want to update both, if you only want to update brunch:
> `sudo chromeos-update -f < path to the Brunch release archive >`

Installed new Brunch release. How can I tell it's been installed correctly? Can I see a version number somewhere from command line ? 

So it's updated correctly -> Chrome OS update tells me I am on the latest version (even with or without options=enable_updates

But the internal mic is not working. 

https://imgur.com/7gwlSXA

it shows in skype for example or other programs. But it's not working.

**Comment by frigis1** [comment]
@sebanc Mic issue still persists - microphone still has no jack detection. The good news is, the laptop no longer shuts down after the welcome screen (note: still using rammus).
@jordanvanbergen One more thing you can try, punch in the following:
sudo nano /etc/modprobe.d/alsa.conf
Then add the following at the end of the file:
options snd-hda-intel model=generic
Then hit ctrl+x, hit Y to save, then reboot.
For me, this disables the headset mic, but enables the internal mic.

**Comment by sebanc** [comment]
@frigis1 ok thanks, i will keep looking into it

**Comment by jordanvanbergen** [comment]
> @sebanc Mic issue still persists - microphone still has no jack detection. The good news is, the laptop no longer shuts down after the welcome screen (note: still using rammus).
> @jordanvanbergen One more thing you can try, punch in the following:
> sudo nano /etc/modprobe.d/alsa.conf
> Then add the following at the end of the file:
> options snd-hda-intel model=generic
> Then hit ctrl+x, hit Y to save, hit enter, then reboot.
> For me, this disables the headset mic, but enables the internal mic.

the sudo nano /etc/modprobe.d/alsa.conf 
with 
options snd-hda-intel model=generic did not work for me (Lenovo Ideapad Yoga 13)

**Comment by sebanc** [comment]
Hi, I applied audio fixes on the new release, not sure which devices were actually affected by those issues though but don't hesitate to try it.

**Comment by nicktelindert** [comment]
@sebanc I'll try it out

**Comment by frigis1** [comment]
@sebanc Thank you, but issue still persists on my laptop.

**Comment by sebanc** [comment]
@frigis1 Thanks for testing, i will keep looking into it

**Comment by tungmeister** [comment]
I'm also getting this issue on my Dell E5840, running `amixer -c0 set "Internal Mic" toggle` does resolve the problem though

**Comment by sebanc** [comment]
@tungmeister you have to create the /var/brunch/bootscripts directory first

**Comment by tungmeister** [comment]
@sebanc I wasn't certain due to the lack of existing brunch directory, I've created the brunch and bootscripts directories however it doesn't work.

**Comment by sebanc** [comment]
Try the following commands:
```
sudo mkdir -p /var/brunch/bootscripts
echo 'amixer -c0 set "Internal Mic" toggle' | sudo tee /var/brunch/bootscripts/alsa.sh
sudo chmod 0755 /var/brunch/bootscripts/alsa.sh
```

**Comment by kbnhvn** [comment]
Hi, same issue for microphone on Dell venue 11 7140. Alsamixer shows a Broadwell rt286 card

**Comment by sebanc** [comment]
For those of you who updated to r83, have you tried the flag "chrome://flags/system-tray-mic-gain" and if so does it help with this issue ?

**Comment by kbnhvn** [comment]
Still no mic on Dell venue 7140, and I lost sound on r83

**Comment by frigis1** [comment]
I have updated to r83 and the latest Brunch framework, and enabled the flag. I only get white noise.

**Comment by sebanc** [comment]
@frigis1 Could you try to update to the latest build in the "brunch-testing" repo:
https://github.com/sebanc/brunch-testing/releases

**Comment by frigis1** [comment]
Updated to latest testing build, still the same result.

**Comment by sebanc** [comment]
Thanks for the feedback on both tests

**Comment by kowalski7cc** [comment]
Same thing happens on XPS 13 9350:
Card: HDA Intel PCH
Chip: Realtek ALC3246
lspci:
1c.5-[3b]----00.0  Realtek Semiconductor Co., Ltd. Device [10ec:525a]

**Comment by sebanc** [comment]
For those who use, 'amixer -c0 set "Internal Mic" toggle' to activate internal mic, could you verify if your mic works without this command if you create a file /etc/init/alsafix.conf (sudo nano /etc/init/alsafix.conf) with the below content ?
```
start on stopped udev-trigger

script
        if [ $(amixer -q -c 0 get "Internal Mic") -eq 0 ]; then amixer -q -c 0 set "Internal Mic" unmute; fi
end script
```

**Comment by frigis1** [comment]
Internal mic does not switch on for me if I use this file.

**Comment by sebanc** [comment]
@frigis1 Thanks! Could you try the below one ?

```
start on stopped udev-trigger and started ui

script
    if [ $(amixer -q -c 0 get "Internal Mic") -eq 0 ]; then amixer -q -c 0 set "Internal Mic" toggle; fi
end script
```

Also if your run "udevadm info" in crosh shell and then plug your headphones, do you see related lines appear in crosh shell ?

**Comment by frigis1** [comment]
That script does not work for me either. I ran udevadm info and the shell listed some options: help, query, path, etc. Didn't seem to provide any information.

**Comment by Shamw0w** [comment]
Hi all, on Dell Inspiron 3000. Internal mic does not work, but headset with mic works when plugged in 3.5mm jack.

I've read the above and tried the following and these are the results I get:

chronos@localhost / $ amixer
Simple mixer control 'Master',0
  Capabilities: pvolume pvolume-joined pswitch pswitch-joined
  Playback channels: Mono
  Limits: Playback 0 - 100
  Mono: Playback 100 [100%] [on]

chronos@localhost / $ amixer -c0 set "Capture" toggle
Simple mixer control 'Capture',0
  Capabilities: cvolume cswitch
  Capture channels: Front Left - Front Right
  Limits: Capture 0 - 63
  Front Left: Capture 63 [100%] [30.00dB] [on]
  Front Right: Capture 63 [100%] [30.00dB] [on]

Even with toggling the mic on and setting capture to 100% I don't seem to capture any input. 

When I run AlsaMixer it only shows 1 master channel (at 100).

Any tips / advice? Have I missed something I need to try in the thread?

**Comment by Shamw0w** [comment]
Also, FWIW, I just did a reboot and tested the mic (without saving a boot script), and the mic worked without needing to toggle it on.

**Comment by hacker1024** [comment]
> @frigis1 Could you try to update to the latest build in the "brunch-testing" repo:
> https://github.com/sebanc/brunch-testing/releases

Enabling the new option added fixes it for me. Thanks for your great work.

**Comment by Adkatka** [comment]
Inspiron 3443 microphone not working ALC3234. 
amixer -c0 set "Internal Mic" toggle wont work
Output: Unable to find Simple Control "Internal Mic"

**Comment by omminp** [comment]
Internal mic not working, but when pluged in headset mic works. Realtek ALC3204 

Fix by crosh shell 
sudo alsactl restore

To make permanent after reboot, 
create /var/brunch/bootscripts as described earlier in this thread, 
create alsa.sh with the line
sudo alsactl.restore
and make alsa.sh executable.

Works for me, so maybe worth a try.

**Comment by dkannegi** [comment]
I am running Chrome 84 Rammus on Brunch r85 k4.19 testing 20200905 on the following laptops: Panasonic Toughbook CF-C2 MK1 and a MK2.  They load up without the mic control showing at all in the UI, my fix was to steal alsa-base.conf from my Linux Mint 20 x64 boot usb and plopped it into Chrome's install at /etc/modprobe.d/alsa.conf (original renamed as a .bak).  One reboot later and I got my mic working as internal mic.  Of note, on the CF-C2s and maybe most other toughbooks, the headphone port is software switchable, however the mic port only shows up as 'internal mic' and an external mic is switched in-line by the jack circuit.  My Amixer -q c0 and c1 output is attached (this computer has two, c1 is my main onboard sound).  
[amixer_c1_panasonic_CFC2MK2.txt](https://github.com/sebanc/brunch/files/5258214/amixer_c1_panasonic_CFC2MK2.txt)
[amixer_c0_panasonic_CFC2MK2.txt](https://github.com/sebanc/brunch/files/5258215/amixer_c0_panasonic_CFC2MK2.txt)
[dmesg.txt](https://github.com/sebanc/brunch/files/5258216/dmesg.txt)
[alsa.conf.txt](https://github.com/sebanc/brunch/files/5258217/alsa.conf.txt)

**Comment by sebanc** [comment]
@dkannegi Thanks for the report, that's a very interesting find :) Now I would need to understand exactly what command is causing this.

I don't really see what in the linux mint alsa-base.conf file could be related to this, however it is replacing the chromeos alsa.conf which contains "options snd cards_limit=1" by default.

Could you rebuild the rootfs and try to only remove "options snd cards_limit=1" from chromeos /etc/modprobe.d/alsa.conf ?

**Comment by dkannegi** [comment]
I upgraded using the chromeos-update command after the first post using ChromeOS 85 and the latest brunch (Brunch r85 k4.19 testing 20200919) which reset the asla.conf and the mic remained available in the UI upon reboot.  I still went into the alsa.conf and #remarked the line in question. Not sure if an upgrade is different from doing a full wipe of the drive.

**Comment by sebanc** [comment]
An upgrade just resets the rootfs partition, several things could have happened here:
- the mic settings remained available because the setting was stored in chromeos partition
- chromeos r85 fixed it.

To directly implement a fix in brunch, I would need to know exactly what modification is necessary to make mic work on a clean install. (Without too many modifications, copying over the full Linux Mint alsa conf over the chromeos would probably have

**Comment by dkannegi** [comment]
I did a test with a fresh USB install of CrOS 84 and brunch r85 k4.19 testing 20200905 with the mic not showing up, and tested commenting out "options snd cards_limit=1".  End result is that the mic still didn't show up.  I will attempt running a chromeos-update command in the USB to bring both CrOS to 85 and brunch to the latest test version and see if it catches it.  Of note,  it turns out this laptop has a Realtek ALC269 (PCIID 10EC:0269) as found out when I put Windows 10 on it, Windows 10 picked it up correctly, but I had another laptop with the same chip not correctly enumerate under Windows so I had basic audio out and no input/mic until I manually  installed the Realtek driver, which then allowed everything to work. Realtek DACs are very weird as they can act as a passthru if there is another sound controller onboard (e.g. nVidia nForce chipsets) OR they can be the sound controller also.

**Comment by sebanc** [comment]
@dkannegi Thanks for trying it :) The issue must then be related to something else, I will try to see what other parameter could cause that

**Comment by omminp** [comment]
@Adkatka I have the same result on my Dell 3185.  
However, I find that 
"amixer -c1 set "Internal Mic" toggle"  works without the error, and then 
"amixer -c1 set "Capture" 40%   also works.  And the mic function is restored.  
Perhaps the device 0 does not exist for this machine.

**Comment by parinay602** [comment]
guys nothing is working....PLease help

**Comment by CreaticaONG** [comment]
Hello,
First of all sorry for my bad English, I didn't practice for a long time...
I have the same issue on my Dell Latitude E5450. I got it working following these steps:

1. My Dell laptop has two sound cards: HDMI and PCH. You can verify your sound cards typing in a shell `aplay -l`
2. In my case HDMI sound card has the 0 id and the PCH has id 1. CRAS (Chrome OS Audio Server) use default id 0 sound card (in my case HDMI that don't have capture function) and that's why the commands above not work for me. To change default sound card I modified  `/etc/modprobe.b/alsa.conf` adding:
`
# Set this to the correct number of cards.
options snd cards_limit=2

options snd-hda-intel id=PCH index=0
options snd-hda-intel id=HDMI index=1
`
3. Save file and reboot.
4. Check the order with `aplay -l`
5.  Perform the steps described above: 
`sudo mkdir /var/brunch/bootscripts
echo 'amixer -c0 set "Internal Mic" toggle' | sudo tee /var/brunch/bootscripts/sound.sh
echo 'amixer -c0 set "Capture" 40%' | sudo tee -a /var/brunch/bootscripts/sound.sh
sudo chmod 0755 /var/brunch/bootscripts/sound.sh
`
Thanks @sebanc for the brunch project!

**Comment by omminp** [comment]
@sebanc 
Folllowing on from the previous comment, I can confirm that in my case also  
aplay -l         **gives**

**** List of PLAYBACK Hardware Devices ****
card 0: HDMI [HDA ATI HDMI], device 3: HDMI 0 [HDMI 0]
  Subdevices: 1/1
  Subdevice #0: subdevice #0
card 1: Generic [HD-Audio Generic], device 0: ALC3204 Analog [ALC3204 Analog]
  Subdevices: 1/1
  Subdevice #0: subdevice #0

**AND**
chronos@localhost ~/Downloads $ amixer -c 0 set "Internal Mic" toggle
amixer: Unable to find simple control 'Internal Mic',0

**AND**
chronos@localhost ~/Downloads $ amixer -c 1 set "Internal Mic" toggle
Simple mixer control 'Internal Mic',0
  Capabilities: cswitch cswitch-joined cswitch-exclusive
  Capture exclusive group: 0
  Capture channels: Mono
  Mono: Capture [on]

With -c 0 the 'set' command returns a null , with the error message.
Perhaps the current internal_mic_fix script could include both cases by checking for a null return from 
amixer -c 0 set "Internal Mic" toggle

and in that case execute
amixer -c 1 set "Internal Mic" toggle

It will mean more users will be fixed.  It seems to be an ongoing issue for users.

**Comment by sebanc** [comment]
@CreaticaONG @omminp Thanks for your investigations and feedback, it is really helpful :)

@omminp I am thinking that forcing the forcing the HDMI soundcard as 2nd might be the proper way to move on with this.
If you set only `options snd-hda-intel id=HDMI index=1` in /etc/modprobe.b/alsa.conf, does it change the soundcard order on your device ?

**Comment by omminp** [comment]
As requested, added 1 option to alsa.conf




*# Set this to the correct number of cards.options snd cards_limit=1options
snd-hda-intel id=HDMI index=1*

The line snd cards_limit=1 was already in the file, and hasn't been changed.

After reboot, the soundcard order is reversed, as here:

chronos@localhost / $ aplay -l
**** List of PLAYBACK Hardware Devices ****
card 0: Generic [HD-Audio Generic], device 0: ALC3204 Analog [ALC3204
Analog]
  Subdevices: 1/1
  Subdevice #0: subdevice #0
card 1: HDMI [HDA ATI HDMI], device 3: HDMI 0 [HDMI 0]
  Subdevices: 1/1
  Subdevice #0: subdevice #0

I hope this helps you work out the best way to proceed.  Let me know if I
can do any further testing for you.




[image: Mailtrack]
<https://mailtrack.io?utm_source=gmail&utm_medium=signature&utm_campaign=signaturevirality5&>
Sender
notified by
Mailtrack
<https://mailtrack.io?utm_source=gmail&utm_medium=signature&utm_campaign=signaturevirality5&>
17/11/20,
06:11:49 pm

**Comment by sebanc** [comment]
@omminp Thanks for the feedback :) I just have a last doubt, could you confirm that with "options=internal_mic_fix" added to the grub config and "options snd-hda-intel id=HDMI index=1" added to /etc/modprobe.b/alsa.conf your microphone works correctly ?

(if the rootfs is rebuilt when you add "options=internal_mic_fix", you will have to add "options snd-hda-intel id=HDMI index=1" to /etc/modprobe.b/alsa.conf again)

**Comment by omminp** [comment]
Well it took a while to realize that I had a file /var/brunch/bootstrap/alsa.sh which I hadn't removed and which enabled the mic...

After deleting that, here are the test results:

[ mic tests.pdf](https://github.com/sebanc/brunch/files/5563336/mic.tests.pdf)

Not sure what the doubt was, but I can confirm that mic works only with both options added.  Tested on 2 installations on the same machine.

**Comment by sebanc** [comment]
@omminp Thanks a lot for the report, it is very clear :)

I have modified the "internal_mic_fix" option accordingly and will add a comment here when I have a new test build with this change included.

**Comment by ulrichunter** [comment]
@CreaticaONG
what do i have to change in alsa.conf to change the sound cards?
Your options don't work for me:
options snd cards_limit=2
options snd-hda-intel id=PCH index=0
options snd-hda-intel id=HDMI index=1

chronos@localhost / $ aplay -l
**** List of PLAYBACK Hardware Devices ****
card 0: Audio [Intel HDMI/DP LPE Audio], device 0: HdmiLpeAudio [Intel HDMI/DP LPE Audi]
  Subdevices: 1/1
  Subdevice #0: subdevice #0
card 0: Audio [Intel HDMI/DP LPE Audio], device 1: HdmiLpeAudio [Intel HDMI/DP LPE Audi]
  Subdevices: 1/1
  Subdevice #0: subdevice #0
card 1: bytcrrt5640 [bytcr-rt5640], device 0: 1 []
  Subdevices: 1/1
  Subdevice #0: subdevice #0
card 1: bytcrrt5640 [bytcr-rt5640], device 1: Deep-Buffer Audio (*) []
  Subdevices: 1/1
  Subdevice #0: subdevice #0

Thank You!

**Comment by sebanc** [comment]
@ulrichunter Could you try "options snd_hdmi_lpe_audio index=1" ?

**Comment by ulrichunter** [comment]
@sebanc the sound cards have changed, but mic is not working... 

chronos@localhost / $ aplay -l
**** List of PLAYBACK Hardware Devices ****
card 0: bytcrrt5640 [bytcr-rt5640], device 0: 1 []
  Subdevices: 0/1
  Subdevice #0: subdevice #0
card 0: bytcrrt5640 [bytcr-rt5640], device 1: Deep-Buffer Audio (*) []
  Subdevices: 1/1
  Subdevice #0: subdevice #0
card 1: Audio [Intel HDMI/DP LPE Audio], device 0: HdmiLpeAudio [Intel HDMI/DP LPE Audi]
  Subdevices: 1/1
  Subdevice #0: subdevice #0
card 1: Audio [Intel HDMI/DP LPE Audio], device 1: HdmiLpeAudio [Intel HDMI/DP LPE Audi]
  Subdevices: 1/1
  Subdevice #0: subdevice #0

Thank You!

**Comment by sebanc** [comment]
@ulrichunter As far as I can tell, now that the sound cards are in the right order and that you have enabled "options=internal_mic_fix", you need to the correct alsa ucm files.

You could try:
https://github.com/alsa-project/alsa-ucm-conf/tree/master/ucm2/bytcr-rt5640

or other ones.

Alsa ucm files go in /usr/share/alsa/ucm directory

**Comment by ulrichunter** [comment]
@sebanc when i add "options=internal_mic_fix" to "grub.cfg" then sound is off also...
i have tryed the original ucm-files and then deleted the original ucm-files and copied the ucm-files from your link to that folder, but mic is not working...

chronos@localhost / $ amixer -c0 set "Internal Mic" toggle
Simple mixer control 'Internal Mic',0
  Capabilities: pswitch pswitch-joined
  Playback channels: Mono
  Mono: Playback [on]

chronos@localhost / $ amixer -c0 set "Capture" 40%
amixer: Unable to find simple control 'Capture',0

Thank You!

**Comment by sebanc** [comment]
Actually, "internal_mic_fix" option just runs `amixer -q -c 0 set "Internal Mic" toggle` on boot, I am not sure why it turns sound off for your device.

Maybe 5.4 kernel might help (I have no clue), I have sent you a github email invite for "brunch-testing" repo.

**Comment by ulrichunter** [comment]
@sebanc I have tryed the testing repo, it's the same... 
maybe CRAS is still using the wrong card, when I start "alsamixer" and hit TAB once to switch to CAPTURE then I see a message "this sound device does not have any capture controls"
when I start alsamixer --card=0 then I have many bars in PLAYBACK and CAPTURE which I can change with the arrow-keys.
Thank You!

**Comment by sebanc** [comment]
There is not really a setup for cras, I believe it is just a virtual interface. Cras might be searching for a specific microphone control name and in your case ucm files might need to be tweaked to be compatible.

Unfortunately, there are many things I still don't understand regarding audio in ChromeOS (even in linux I don't have much audio framework knowledge) so I cannot really help further with this.

**Comment by ulrichunter** [comment]
@sebanc I can live without a mic... it's a great framework! Thank You!

**Comment by k0gen** [comment]
On Lenovo x270 and older C50-30 I have found out that: 
`options snd_hda_intel index=0 model=headset-mic` in /etc/modprobe.d/alsa.conf fixes muted headphone jack. How can we make it permanent so we don’t have to reapply it after changing and trying some other options in grub?

**Comment by sebanc** [comment]
@k0gen Adding "snd_hda_intel.index=0 snd_hda_intel.model=headset-mic" on the kernel command line should work the same and be robust against updates.

**Comment by k0gen** [comment]
> @k0gen Adding "snd_hda_intel.index=0 snd_hda_intel.model=headset-mic" on the kernel command line should work the same and be robust against updates.
Just did that and can confirm this works perfectly! Thank You. (I'll go and update my wiki entry regarding lenovo x270 now)

**Comment by ludem** [comment]
> 
> 
> @sebanc the sound cards have changed, but mic is not working...
> 
> chronos@localhost / $ aplay -l
> **** List of PLAYBACK Hardware Devices ****
> card 0: bytcrrt5640 [bytcr-rt5640], device 0: 1 []
> Subdevices: 0/1
> Subdevice #0: subdevice #0
> card 0: bytcrrt5640 [bytcr-rt5640], device 1: Deep-Buffer Audio (*) []
> Subdevices: 1/1
> Subdevice #0: subdevice #0
> card 1: Audio [Intel HDMI/DP LPE Audio], device 0: HdmiLpeAudio [Intel HDMI/DP LPE Audi]
> Subdevices: 1/1
> Subdevice #0: subdevice #0
> card 1: Audio [Intel HDMI/DP LPE Audio], device 1: HdmiLpeAudio [Intel HDMI/DP LPE Audi]
> Subdevices: 1/1
> Subdevice #0: subdevice #0
> 
> Thank You!

Same card here: internal mic doesn't work and headset is not recognized when plugged in and the audio doesn't switch.
I've tried suggested kernel options with no luck.

**Comment by xellotath** [comment]
@sebanc So I don't know a lot about the internals but I played with this for a while and did some testing on my Lenovo Yoga c940, and found some things: 

arecord -L returns only one card, sof-hda-dsp. alsamixer -c0 reports the chip as Realtek ALC298.
arecord -l returns 4 devices(it used to be only HDA Analog, but after some tweaking in the options, including  disable_intel_hda, got them all to show up). The rest are now called HDA Digital, DMIC, and DMIC16kHz.

alsamixer -c0 showed however the ones that appeared under amixer -c0 scontents that had capture channels under them. Those were Mic Boost, Capture, Dmic0, Dmic1 2nd, PGA2.0 2 Master, and PGA4.0 4 Master. Using mic-test.com I could see whether there was input and from where by playing with their volumes on alsamixer. Out of all these I should mention, only Capture and Dmic could toggle through amixer -c0 set [...] toggle. Interestingly enough amixer also reported a device called 'Mic' that has playback but no capture channels. Toggling it on or changing its volume did nothing.

Anyway, I came down to the following: Capture records the audio jack for me. It produces white noise until you connect headphones, after which it records the sounds played back on them. Dmic is my actual mic, but I could only tell after booting in mint where I could see changes in alsamixer reflected the mic's(Dmic) settings(it's properly recognized there, and everything is identical). PGA2.0 2 Master is the volume of the headphone microphone, when there is one connected. It cannot be toggled. Toggling the ones that *can* be toggled does not disable the headphone mic.

So basically I know my mic is the Dmic device on my sound card, but the one being used is the wrong device. And I don't really know how to change it.

**Comment by JamesMaria** [comment]
> @k0gen Adding "snd_hda_intel.index=0 snd_hda_intel.model=headset-mic" on the kernel command line should work the same and be robust against updates.

How to open kernel command line?

**Comment by JamesMaria** [comment]
@sebanc How to open up kernel lines and also any more updates on fixing microphone

**Comment by omminp** [comment]
I found the laptop mic had stopped working, it had responded to the internal_mic_fix when I applied it earlier.  I found a couple of things that may be worth updating, since internal mic operation still seems to cause problems for some users.
1. I need the option "internal_mic_fix2" for my laptop
2. that option is not listed under 'Framework options' in the Readme.md   - should be added there
3. that option is not listed in the setup routine in GRUB  - should be added there
4. editing the GRUB at boot now shows 'options=$options' rather than the options that have been set, additional options added after and separated by a comma are executed after whatever is listed in $options  .  This needs to be explained in the paragraph at the end of 'Framework options' .

I was able to add 'internal_mic_fix2' to GRUB and see it added in dmesg after booting - mic then operates correctly for me.

**Comment by channelup69** [comment]
> mod

Did you ever get this resolved I have a very similar set up

**Comment by omminp** [comment]
the option for "internal_mic_fix2" has been reinstated to the Brunch setup screen and works for me.

**Comment by RuBronyCone** [comment]
The only way I'm going to get the microphone to work properly on the Lenovo 330 is to place the script in the usr/local/bin directory, grant permissions, and run it through a shell. Here is the script: 
`arecord -f cd -d 5 /tmp/test-mic.wav
amixer -c0 set "Capture" 0%,100% unmute cap`
When I put this script in var/brunch/bootscript and give it the right permissions, nothing happens when I restart it. Either this sequence does not work at the stage at which the bootscripts work, or I'm doing something wrong. How can I force this script to run on login?
